homeassistant/components/automation/__init__.py

```diff
@@ -374,52 +496,73 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
 
         This method is a coroutine.
         """
-        if self._variables:
-            try:
-                variables = self._variables.async_render(self.hass, run_variables)
-            except template.TemplateError as err:
-                self._logger.error("Error rendering variables: %s", err)
+        reason = ""
+        if "trigger" in run_variables and "description" in run_variables["trigger"]:
+            reason = f' by {run_variables["trigger"]["description"]}'
+        self._logger.debug("Automation triggered%s", reason)
+
+        trigger = run_variables["trigger"] if "trigger" in run_variables else None
+        with trace_automation(
+            self.hass, self.unique_id, self._raw_config, trigger, context
+        ) as automation_trace:
+            if self._variables:
+                try:
+                    variables = self._variables.async_render(self.hass, run_variables)
+                except template.TemplateError as err:
+                    self._logger.error("Error rendering variables: %s", err)
+                    automation_trace.set_error(err)
+                    return
+            else:
+                variables = run_variables
+            automation_trace.set_variables(variables)
+
+            if (
+                not skip_condition
```

Comment:
We should store in the trace if we're skipping the condition.

Reply:
This only happens if we're triggered by a service call. Should we attach the service call data, except the variables, to the trace?